server/src/main/java/org/jboss/as/server/deployment/DeploymentAddHandler.java

```diff
@@ -230,6 +231,25 @@ private static String getRuntimeName(String name, ModelNode deployment) {
         return new DeploymentHandlerUtil.ContentItem(hash);
     }
 
+    private InputStream doTransformation(OperationContext context, ModelNode contentItemNode, String name, InputStream in) throws IOException, OperationFailedException {
+        try {
+            return deploymentTransformer.transform(in, name);
+        } catch (RuntimeException t) {
+            // Check if the InputStream is already attached to the operation request (as per CONTENT_INPUT_STREAM_INDEX check) and ignore that case
+            // as calling getInputStream would of returned the already partially consumed InputStream.
+            // Also verify that the thrown exception is the specific WFCORE-5198 `Error code 3`.
+            if (!contentItemNode.hasDefined(DeploymentAttributes.CONTENT_INPUT_STREAM_INDEX.getName()) &&
+                    t.getCause() != null && t.getCause().getCause() != null &&
+                    t.getCause().getCause() instanceof IOException &&
+                    t.getCause().getCause().getMessage().contains("during transformation. Error code 3")) {
```

Comment:
I think it will be ok to retry with the untransformed deployment no matter what the exception was but it's ok to start off being more restrictive.